File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(count)
    @count = count
  end
  def square_of_sum
    (1..@count).sum ** 2
  end
  def sum_of_squares
    (1..@count).sum { |n| n ** 2 }
  end
  def difference
    self.square_of_sum - self.sum_of_squares
  end
end

File: swift/difference-of-squares/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
    var num: Int

    init(_ num: Int) {
        self.num = num
    }

    lazy var squareOfSum: Int = {
        let sum = (1...(self.num)).reduce(0, +)
        return sum * sum
    }()

    lazy var sumOfSquares: Int = {
        var total = 0
        for i in 1...self.num {
            total += i*i
        }
        return total
    }()

    lazy var differenceOfSquares: Int = {
        self.squareOfSum - self.sumOfSquares
    }()
}
